generator/content.py

```diff
@@ -0,0 +1,146 @@
+"""CLI for generating content folders for Looker namespaces."""
+import logging
+import os
+
+import click
+import looker_sdk
+import yaml
+
+
+def _setup_env_with_looker_creds():
+    client_id = os.environ["LOOKER_API_CLIENT_ID"]
+    client_secret = os.environ["LOOKER_API_CLIENT_SECRET"]
+    instance = os.environ["LOOKER_INSTANCE_URI"]
+
+    os.environ["LOOKERSDK_BASE_URL"] = instance
+    os.environ["LOOKERSDK_API_VERSION"] = "3.1"
+    os.environ["LOOKERSDK_VERIFY_SSL"] = "true"
+    os.environ["LOOKERSDK_TIMEOUT"] = "120"
+    os.environ["LOOKERSDK_CLIENT_ID"] = client_id
+    os.environ["LOOKERSDK_CLIENT_SECRET"] = client_secret
+
+
+def generate_folders(namespaces: dict):
+    """Generate folders and ACLs for namespaces."""
+    sdk = looker_sdk.init31()
+    logging.info("Looker SDK 3.1 initialized successfully.")
+
+    shared_folders = sdk.search_folders(name="home")
+    if len(shared_folders) != 1:
+        raise click.ClickException(
+            f"Found wrong number of shared folders: {shared_folders}"
+        )
+
+    shared_folder_id = shared_folders[0].id
+    if shared_folder_id is None:
+        raise click.ClickException("Error: Shared folder id is missing")
```

Comment:
this pattern is used so many times, I think it should be a separate function that is passed a list and returns an id

Reply:
Makes sense, thanks for the suggestion!